guidance/03-analysis/analysis-steps/02-analysis-modelling.qmd

```diff
@@ -17,39 +73,40 @@ Starting from the source tables, we want to create the agreed outputs. Depending
-    * [Government Functional Standard (publishing.service.gov.uk)](https://assets.publishing.service.gov.uk/media/64e4cf0a4002ee000d560d5f/2023.02_GovS_010_Analysis_v2.1.pdf)
-
 ## Tips for effective programming
+
+Data will be a factor in all initiatives. Most initiatives will also have some element of coding involved, even when using low-or-no-code tools like Power BI.
+
 * Use Github to handle version control, issues and code reviews.
 * In Python, use object-oriented programming (OOP) if you can. Please see the internal Data Science Wiki for information.
 * Although R does have support for OOP, it is not a big paradigm like in Python.
 * Be careful of premature optimisation (called out by computer scientist Donald Knuth as ‘the root of all evil’ in software development).
 * Start small, such as limiting data to a manageable subset when developing the analysis; this makes it easier to spot issues and allows for faster iteration of code.
 * Check the data looks correct at each step of reshaping and transforming it; better still, know what it should be before writing and applying the step (the principle behind test driven development, TDD).
-* Work on one component of output at a time, for example using `shiny` with the modular `golem` framework, as in our [template](https://github.com/nhsbsa-data-analytics/nhsbsaShinyR).
-* Frequently run code as you write it, so any issues are encountered early and their source is clearer.
+* Work on one component of output at a time, for example using `shiny` with the modular `golem` framework, as in our [shiny app template](https://github.com/nhsbsa-data-analytics/nhsbsaShinyR).
+* Frequently run code as you write it, or check the output for low-or-no-code tools, so any issues are encountered early and their source is clearer.
 * You may produce multiple smaller data sets to be used in your outputs; the considerations applying to base tables apply equally to these.
 * Where appropriate, the final code should use RAP principles [LINK?].
 
 ## Check the quality
+
 As with data preparation, it is important to test and validate your results.
 
 * Where possible, check results against other sources.
 * Other sources could be data from previous initiatives, Management Information dashboards or ePACT2.
 
 Additionally, you must be careful about visualising sensitive results. Please contact your critical friend or the Data Science Lead if you are unsure. Results for external audience or general public must be subjected to Statistical Disclosure Control [LINK] before release.
-Before pushing the code to GitHub, it's essential to ensure that the output is cleared, and that no sensitive information remains within it.
 
-### RAP considerations (LINK?)
+Before pushing any code to GitHub, it's essential to ensure that the output is cleared, and that no sensitive information remains within it.
+
+## RAP considerations (LINK?)
```

Comment:
what are we doing about a RAP link?

Reply:
Added useful RAP resources section with couple of links